src/sorcha/sorcha.py

```diff
@@ -332,16 +324,19 @@ def runLSSTSimulation(args, configs):
             observations.reset_index(drop=True, inplace=True)
             verboselog("Number of rows AFTER applying SSP linking filter: " + str(len(observations.index)))
 
-        pplogger.info("Post processing completed for this chunk")
-
-        pplogger.info("Output results for this chunk")
-
-        # write output
-
-        PPWriteOutput(args, configs, observations, endChunk, verbose=args.verbose, lastchunk=lastChunk)
+        # write output if chunk not empty
+        if len(observations.index) > 0:
+            pplogger.info("Post processing completed for this chunk")
+            pplogger.info("Outputting results for this chunk")
+            PPWriteOutput(args, configs, observations, endChunk, verbose=args.verbose)
+            if args.stats is not None:
+                stats(observations, args.stats, args.outpath, linking=configs["SSP_linking_on"])
+        else:
+            verboselog("No observations left in chunk. No output will be written for this chunk.")
 
-        if args.stats is not None:
-            stats(observations, args.stats, args.outpath, linking=configs["SSP_linking_on"])
+        if endChunk >= lenf and configs["output_format"] == "sqlite3":
+            pplogger.info("Last chunk detected. Indexing output SQLite database...")
```

Comment:
Why not pull this outside of the loop rather than check if on the last chunk. If the loop completed you finished the last  chunk and can index. Saves a comparison.

Reply:
I think you need to check the file exists because if it doesn't because all the chunks were empty there is no file. I'm not sure what's the best place for that.